src/main/java/seedu/address/model/module/UniqueModuleList.java

```diff
@@ -0,0 +1,127 @@
+ * as to ensure that the person with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Module#isSameModule(Module)
+ */
+public class UniqueModuleList implements Iterable<Module> {
+
+    private final ObservableList<Module> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Module> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent module as the given argument.
+     */
+    public boolean contains(Module toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameModule);
+    }
+
+    /**
+     * Adds a module to the list.
+     * The module must not already exist in the list.
+     */
+    public void add(Module toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateModuleException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Removes the equivalent module from the list.
+     * The module must exist in the list.
+     */
+    public void remove(Module toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ModuleNotFoundException();
+        }
+    }
+
+    /**
+     * Gets the module with {@code code} from the list.
+     * The module must exist in the list.
+     */
+    public Module getModule(String code) {
+        requireNonNull(code);
+        for (Module mod : internalList) {
+            if (mod.getCode().equals(code)) return mod;
+        }
+        throw new ModuleNotFoundException();
+    }
+
+    /**
+     * Replaces the contents of this list with {@code modules}.
+     * {@code modules} must not contain duplicate modules.
+     */
+    public void setModules(List<Module> modules) {
+        requireAllNonNull(modules);
+        if (!modulesAreUnique(modules)) {
+            throw new DuplicateModuleException();
+        }
+
+        internalList.setAll(modules);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Module> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Module> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof UniqueModuleList // instanceof handles nulls
+                        && internalList.equals(((UniqueModuleList) other).internalList));
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    /**
+     * Returns true if {@code modules} contains only unique modules.
+     */
+    private boolean modulesAreUnique(List<Module> modules) {
```

Comment:
Would it be better as `areModulesUnique`, following the format of the convention of `isFound`